### PR TITLE
Attempt to ship logger errors via the CloudHandler

### LINKED ISSUE #1943: Attempt to log possible log writing errors
## Current behavior
Currently if the logger encounters an error it isn't immediately surfaced to Cloud users what the reason is for the logging error.

## Proposed behavior
Something along the lines of (but not limited to):
```python
try:
   write_some_logs()
except Exception as exc:
    sleep(1)
    write_this_error_log(exc)
except Exception as exc2:
    raise exc    # raise exception as it normally would
```

Where if the writing of the logs fails then the exception is passed as an attempted log. Otherwise if the exception logging fails then raise as it normally would. This would give some nice insight into "missing" logs.

## Example
Possible issues such as non serializeable bytes, large payloads, etc. could be surfaced

```
self.logger.info(b"bad bytes string")
CRITICAL - CloudHandler | Failed to write log with error: Object of type bytes is not JSON serializable
```

```
self.logger.info("LARGE STRING...........")
CRITICAL - CloudHandler | Failed to write log with error: 413 Client Error: Request Entity Too Large for url: https://api.prefect.io/graphql/alpha
```

### PR DESCRIPTION
**Pull request by joshmeek**
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [x] adds new tests (if appropriate)
- [x] updates `CHANGELOG.md` (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Note that your PR will not be reviewed unless all three boxes are checked.

## What does this PR change?
Adds a step to the CloudHandler to attempt to log possible logger errors to Cloud for visibility. For example if the payload is too large or something non-json serializable is attempted the error will be raised in the flow process but not elevated for visibility.

Example logger issue being elevated as a cloud log:
image 

Closes #1943 

## Why is this PR important?
Improves user visibility into otherwise hidden logger exceptions.

### CODE REVIEW

File: src/prefect/utilities/logging.py [hunk 1]
```diff
@@ -92,7 +98,10 @@ def put(self, log: dict) -> None:
             json.dumps(log)  # make sure the payload is serializable
             self.queue.put(log)
         except TypeError as exc:
-            self.logger.critical("Failed to write log with error: {}".format(str(exc)))
+            message = "Failed to write log with error: {}".format(str(exc))
+            self.logger.critical(message)
+
+            self.queue.put(self._make_error_log(message))
```
Comment (cicdw) [inline src/prefect/utilities/logging.py]: Could this result in an infinite loop if, heaven forbid, Cloud was completely down for an extended period of time?

File: src/prefect/utilities/logging.py [hunk 2]
```diff
@@ -125,7 +134,21 @@ def emit(self, record) -> None:  # type: ignore
             log["info"] = record_dict
             self.put(log)
         except Exception as exc:
-            self.logger.critical("Failed to write log with error: {}".format(str(exc)))
+            message = "Failed to write log with error: {}".format(str(exc))
+            self.logger.critical(message)
+
+            self.put(self._make_error_log(message))
```
Comment (cicdw) [inline src/prefect/utilities/logging.py]: Same question as above

File: src/prefect/utilities/logging.py (hunk 1 above)
Comment (joshmeek) [reply src/prefect/utilities/logging.py]: Cloud errors like 400s are surfaced in the above handle. Where an extra check happens  and logs `Unable to write logs to Prefect Cloud`

Comment (joshmeek) [reply src/prefect/utilities/logging.py]: Let me test this again but I believe that's what should happen

**Comment by joshmeek** [comment]
@cicdw Tested this some more and we should be good to go 👍 the only potential _issue_ is the batch upload error logging part:
```python
    def batch_upload(self) -> None:
        logs = []
        try:
            while True:
                log = self.queue.get(False)
                logs.append(log)
        except Empty:
            pass

        if logs:
            try:
                assert self.client is not None
                self.client.write_run_logs(logs)
            except Exception as exc:
                message = "Failed to write log with error: {}".format(str(exc))
                self.logger.critical(message)

                # Attempt to write batch error log otherwise log invalid cloud communication
                try:
                    assert self.client is not None
                    self.client.write_run_logs([self._make_error_log(message)])
                except Exception as exc:
                    self.logger.critical("Unable to write logs to Prefect Cloud")
```

If there's an issue with one of the logs in this part e.g. `Failed to write log with error: 413 Client Error: Request Entity Too Large` then since we are writing all of the logs at once they all fail. Thoughts?

**Pull request merged**